packages/browser/src/util/urlPath.spec.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { appendToUrlPathname } from "./urlPath";
+
+describe("urlPath", () => {
+  it("should remove one slash if empty path", () => {
+    expect(appendToUrlPathname("https://ex.com/", "/test")).toEqual(
+      "https://ex.com/test"
+    );
+
+    expect(
+      appendToUrlPathname("https://ex.com/", "////only-remove-one-slash")
+    ).toEqual("https://ex.com////only-remove-one-slash");
```

Comment:
Is this a behaviour we actually want ?

Reply:
Well, good question - but really this is a consequence of whoever wrote this utility function not writing comprehensive tests at the time they wrote it - i.e., I was just writing tests to cover what the code was actually doing. I'm assuming that since the code 'at large' seems to work fine, that this code is in fact doing the 'right thing'.
But I'm pretty sure it is **_actually_** the right thing to do too (i.e., if the 'path' and the 'append-to-path' strings end and start, respectively, with slashes, then remove one of them when concatenating) - seems to makes sense to me anyways.